rust-bins/src/bin/keygen-genesis.rs

```diff
@@ -0,0 +1,241 @@
+use clap::AppSettings;
+use client_server_helpers::*;
+use crypto_common::*;
+use curve_arithmetic::Curve;
+use elgamal::PublicKey;
+use id::types::*;
+use std::convert::TryFrom;
+
+use pairing::bls12_381::Bls12;
+use sha2::Sha512;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use curve25519_dalek::edwards::CompressedEdwardsY;
+use ed25519_dalek::*;
+use pairing::bls12_381::{G1, G2};
+use std::fs;
+#[derive(StructOpt)]
+struct KeygenIp {
+    #[structopt(long = "rand-input", help = "File with randomness.")]
+    rand_input: PathBuf,
+    #[structopt(
+        long = "ip-identity",
+        help = "The integer identifying the identity provider"
+    )]
+    ip_identity: u32,
+    #[structopt(long = "name", help = "Name of the identity provider")]
+    name: String,
+    #[structopt(long = "url", help = "url to identity provider")]
+    url: String,
+    #[structopt(long = "description", help = "Description of identity provider")]
+    description: String,
+    #[structopt(
+        long = "bound",
+        help = "Upper bound on messages signed by the IP",
+        default_value = "30"
+    )]
+    bound: u32,
+    #[structopt(long = "out-pub", help = "File to output the public keys to.")]
+    out_pub: PathBuf,
+}
+
+#[derive(StructOpt)]
+struct KeygenAr {
+    #[structopt(long = "rand-input", help = "File with randomness.")]
+    rand_input: PathBuf,
+    #[structopt(
+        long = "ar-identity",
+        help = "The integer identifying the anonymity revoker"
+    )]
+    ar_identity: u32,
```

Comment:
Do we have a `FromStr` instance for `ArIdentity`? If so we should use ArIdentity here directly. IF not, disregard this comment, but we do need to make sure this is not 0.